File: Basic/Python/1006.py
'''
@Author: kurumi
@Date: 2025-09-06
@Description: 1006 换个格式输出整数
'''
def toB(n: int) -> str:
    return "B" * n

def toS(n: int) -> str:
    return "S" * n

def toN(n: int) -> str:
    res: str = ""
    for i in range(n):
        res += str(i + 1)
    return res

def main():
    res: str = ""
    n: int = int(input())
    res += toB(n // 100)
    n %= 100
    res += toS(n // 10)
    n %= 10
    res += toN(n)
    print(res)

if __name__ == "__main__":
    main()

File: Basic/JS/1006.js
/**
 * @author: kurumi
 * @date: 2025-09-06
 * @description: 1006 换个格式输出整数
 */
const rl = require("readline").createInterface({input: process.stdin});
const iter = rl[Symbol.asyncIterator]();
const readline = async () => (await iter.next()).value;

const toB = n => "B".repeat(n);
const toS = n => "S".repeat(n);
const toN = n => {
  let res = "";
  for (let i = 1; i <= n; i++) res += i;
  return res;
}

void (async function() {
  let str = "";
  let n = parseInt(await readline());
  str += toB(Math.floor(n / 100));
  n %= 100;
  str += toS(Math.floor(n / 10));
  n %= 10;
  str += toN(n);
  console.log(str);
})();